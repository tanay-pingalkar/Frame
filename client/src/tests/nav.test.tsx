import { Provider } from "react-redux";
import configureMockStore, { MockStoreEnhanced } from "redux-mock-store";
import thunk from "redux-thunk";
import { cleanup, render, screen, act } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import Nav from "../components/nav";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

const mockStore = configureMockStore([thunk]);

describe("testing frame component", () => {
  let store: MockStoreEnhanced<unknown>;
  beforeEach(() => {
    store = mockStore({
      userInfo: {
        name: "a name",
      },
      isAuth: true,
    });
  });
  afterEach(() => {
    jest.clearAllMocks();
    cleanup();
  });
  it("should render Nav properly", async () => {
    const history = createMemoryHistory();
    const pushSpy = jest.spyOn(history, "replace");
    render(
      <Provider store={store}>
        <Router history={history}>
          <Nav></Nav>
        </Router>
        ,
      </Provider>
    );
    const btn = screen.getAllByText("Post frame");
    const toggler = screen.getByTestId("toggler");
    expect(toggler.childElementCount).toBe(8);
    expect(screen.getAllByText("Home").length).toBe(1);
    expect(screen.getAllByText("Notification").length).toBe(1);
    expect(screen.getAllByText("a name").length).toBe(1);
    expect(screen.getAllByText("Search").length).toBe(1);
    expect(btn.length).toBe(1);
    act(() => {
      screen.getByTestId("button").click();
    });

    expect(pushSpy).toBeCalledWith("/app/upload");
    console.log();
  });
  it("should render Nav when logging is false", () => {
    store = mockStore({
      userInfo: {},
      isAuth: "loggin",
    });
    render(
      <Provider store={store}>
        <Nav></Nav>
      </Provider>
    );
    expect(screen.getAllByText("loggin").length).toBe(1);
  });
  it("should match snapshot", () => {
    const snapshot = renderer.create(
      <Provider store={store}>
        <Nav></Nav>
      </Provider>
    );
    expect(snapshot).toMatchSnapshot();
  });
});
