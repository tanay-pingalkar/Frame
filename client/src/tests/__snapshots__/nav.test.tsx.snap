// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing frame component should match snapshot 1`] = `
<div
  className="nav"
>
  <div
    className="profile"
  >
    <img
      alt="profile"
      src="Wallpaper-Malavida-Green-Top-Image.jpg"
    />
    <h2>
      a name
    </h2>
  </div>
  <div
    className="toggler"
    data-testid="toggler"
  >
    <svg
      fill="none"
      height="22"
      viewBox="0 0 22 22"
      width="22"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21.2971 20.0385L16.0611 14.5928C17.4074 12.9924 18.145 10.9788 18.145 8.88249C18.145 3.98476 14.1602 0 9.26251 0C4.36478 0 0.38002 3.98476 0.38002 8.88249C0.38002 13.7802 4.36478 17.765 9.26251 17.765C11.1012 17.765 12.8534 17.2104 14.3514 16.1576L19.6272 21.6447C19.8477 21.8737 20.1443 22 20.4622 22C20.763 22 21.0484 21.8853 21.2651 21.6768C21.7254 21.2338 21.7401 20.4992 21.2971 20.0385ZM9.26251 2.31717C12.8827 2.31717 15.8278 5.2623 15.8278 8.88249C15.8278 12.5027 12.8827 15.4478 9.26251 15.4478C5.64232 15.4478 2.69719 12.5027 2.69719 8.88249C2.69719 5.2623 5.64232 2.31717 9.26251 2.31717Z"
        fill="black"
      />
    </svg>
    <h2>
      Search
    </h2>
    <br />
    <svg
      fill="none"
      height="27"
      onClick={[Function]}
      viewBox="0 0 27 27"
      width="27"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clipPath="url(#clip0)"
      >
        <path
          d="M26.325 11.691C26.3242 11.6904 26.3235 11.6895 26.3229 11.6889L15.309 0.675454C14.8396 0.205788 14.2154 -0.0527344 13.5515 -0.0527344C12.8876 -0.0527344 12.2634 0.205788 11.7938 0.675454L0.785644 11.6834C0.781936 11.6871 0.778022 11.691 0.77452 11.6947C-0.189532 12.6643 -0.187884 14.2375 0.779258 15.2046C1.22112 15.6467 1.80449 15.9025 2.42845 15.9295C2.45399 15.932 2.47953 15.9332 2.50528 15.9332H2.94405V24.0383C2.94405 25.6423 4.24923 26.9473 5.8533 26.9473H10.1623C10.5992 26.9473 10.9533 26.593 10.9533 26.1563V19.8018C10.9533 19.0699 11.5488 18.4746 12.2807 18.4746H14.8223C15.5542 18.4746 16.1495 19.0699 16.1495 19.8018V26.1563C16.1495 26.593 16.5036 26.9473 16.9405 26.9473H21.2495C22.8538 26.9473 24.1588 25.6423 24.1588 24.0383V15.9332H24.5658C25.2295 15.9332 25.8537 15.6747 26.3235 15.2048C27.2917 14.2362 27.2921 12.6602 26.325 11.691V11.691ZM25.2048 14.0863C25.034 14.2571 24.807 14.3512 24.5658 14.3512H23.3677C22.9308 14.3512 22.5767 14.7053 22.5767 15.1422V24.0383C22.5767 24.77 21.9814 25.3653 21.2495 25.3653H17.7315V19.8018C17.7315 18.1977 16.4266 16.8925 14.8223 16.8925H12.2807C10.6764 16.8925 9.37127 18.1977 9.37127 19.8018V25.3653H5.8533C5.12161 25.3653 4.52608 24.77 4.52608 24.0383V15.1422C4.52608 14.7053 4.17198 14.3512 3.73507 14.3512H2.5576C2.54525 14.3504 2.53309 14.3497 2.52053 14.3495C2.28487 14.3454 2.06384 14.2519 1.89822 14.0861C1.54597 13.7338 1.54597 13.1605 1.89822 12.8081C1.89842 12.8081 1.89842 12.8079 1.89863 12.8077L1.89925 12.8071L12.9127 1.794C13.0833 1.62323 13.3101 1.5293 13.5515 1.5293C13.7927 1.5293 14.0195 1.62323 14.1903 1.794L25.2013 12.8048C25.2029 12.8064 25.2048 12.8081 25.2064 12.8097C25.5568 13.1626 25.5562 13.7346 25.2048 14.0863V14.0863Z"
          fill="black"
        />
      </g>
      <defs>
        <clipPath
          id="clip0"
        >
          <rect
            fill="white"
            height="27"
            width="27"
          />
        </clipPath>
      </defs>
    </svg>
    <h2>
      Home
    </h2>
    <br />
    <svg
      fill="none"
      height="27"
      viewBox="0 0 22 27"
      width="22"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21.5458 21.4868C21.4641 21.4353 20.849 21.0168 20.2331 19.7509C19.102 17.4263 18.8646 14.1516 18.8646 11.8138C18.8646 11.8036 18.8644 11.7936 18.864 11.7834C18.8516 8.69379 16.9465 6.03039 14.2203 4.83537V3.00776C14.2203 1.34926 12.8331 0 11.1281 0H10.8719C9.16685 0 7.77972 1.34926 7.77972 3.00776V4.83527C5.04435 6.03413 3.13542 8.71114 3.13542 11.8138C3.13542 14.1516 2.898 17.4262 1.76692 19.7509C1.15106 21.0168 0.535967 21.4352 0.454186 21.4868C0.109794 21.6416 -0.0600133 21.9991 0.0192154 22.3614C0.0992043 22.7274 0.451417 22.9796 0.836048 22.9796H6.80655C6.83989 25.2019 8.7076 27 11 27C13.2924 27 15.1602 25.2019 15.1935 22.9796H21.164C21.5486 22.9796 21.9008 22.7274 21.9808 22.3614C22.0599 21.9991 21.8902 21.6415 21.5458 21.4868ZM9.40888 3.00776C9.40888 2.22302 10.0652 1.58462 10.8719 1.58462H11.1282C11.9349 1.58462 12.5912 2.22302 12.5912 3.00776V4.32174C12.0771 4.21854 11.5448 4.16422 10.9998 4.16422C10.4549 4.16422 9.92286 4.21849 9.40893 4.32158V3.00776H9.40888ZM11 25.4154C9.60589 25.4154 8.46872 24.3282 8.43565 22.9797H13.5643C13.5313 24.3281 12.3942 25.4154 11 25.4154ZM14.2799 21.395C14.2798 21.395 2.69241 21.395 2.69241 21.395C2.83333 21.1817 2.97674 20.9402 3.11929 20.6673C4.21101 18.5768 4.76458 15.598 4.76458 11.8138C4.76458 8.46956 7.56169 5.74884 10.9997 5.74884C14.4378 5.74884 17.2349 8.46956 17.2349 11.8165C17.2349 11.8262 17.2351 11.8359 17.2355 11.8456C17.2386 15.6146 17.7921 18.5827 18.8808 20.6673C19.0233 20.9403 19.1668 21.1817 19.3076 21.395H14.2799Z"
        fill="black"
      />
      <circle
        cx="17"
        cy="6"
        fill="black"
        r="5"
      />
    </svg>
    <h2>
      Notification
    </h2>
  </div>
  <div
    className="add"
    data-testid="button"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="22"
      viewBox="0 0 40 40"
      width="22"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M36.4286 16.4286H24.2857C23.8913 16.4286 23.5714 16.1087 23.5714 15.7143V3.57143C23.5714 1.59912 21.9723 0 20 0C18.0277 0 16.4286 1.59912 16.4286 3.57143V15.7143C16.4286 16.1087 16.1087 16.4286 15.7143 16.4286H3.57143C1.59912 16.4286 0 18.0277 0 20C0 21.9723 1.59912 23.5714 3.57143 23.5714H15.7143C16.1087 23.5714 16.4286 23.8913 16.4286 24.2857V36.4286C16.4286 38.4009 18.0277 40 20 40C21.9723 40 23.5714 38.4009 23.5714 36.4286V24.2857C23.5714 23.8913 23.8913 23.5714 24.2857 23.5714H36.4286C38.4009 23.5714 40 21.9723 40 20C40 18.0277 38.4009 16.4286 36.4286 16.4286Z"
        fill="black"
      />
    </svg>
    <h2>
      Post frame
    </h2>
  </div>
</div>
`;
